domain/modeldefaults/state/state.go

```diff
@@ -167,7 +168,7 @@ WHERE m.uuid = ?
 	return map[string]string{
 		config.NameKey: modelName,
 		config.UUIDKey: uuid.String(),
-		config.TypeKey: modelType,
+		config.TypeKey: cloudType,
 	}, nil
```

Comment:
Can we add a big TODO that this needs to die with fire (and add to risk register if not there).

Reply:
What exactly needs to die with fire?

Reply:
the whole method - it only exists because model metadata is currently stored in model config

Reply:
There's already a TODO on the call site:
https://github.com/juju/juju/blob/58cb44cdf5e0c898e1646c08df76e3f8b33cfe98/domain/modeldefaults/service/service.go#L138-L146